lib/model.js

```diff
@@ -3575,13 +3575,17 @@ Model.bulkWrite = function(ops, options, callback) {
         return cb(null, getDefaultBulkwriteResult());
       }
 
-      this.$__collection.bulkWrite(ops, options, (error, res) => {
-        if (error) {
-          return cb(error);
-        }
+      try {
+        this.$__collection.bulkWrite(ops, options, (error, res) => {
+          if (error) {
+            return cb(error);
+          }
 
-        cb(null, res);
-      });
+          cb(null, res);
+        });
+      } catch (err) {
+        return cb(err);
```

Comment:
I've always didn't like the fact that bulkWrite could throw both sync and async errors, but I accepted it as a fact.
Glad this is changing now.